tests/e2e/test_access.py

```diff
@@ -0,0 +1,126 @@
+import os
+import unittest
+
+# requests is not explicitly installed system wide in our requirements.txt
+# (.setup/pip/system_requirements.txt), but exists currently installed
+# in the vagrant vm as required by other packages
+import requests
+
+CURRENT_PATH = os.path.dirname(os.path.realpath(__file__))
+
+
+class TestAccess(BaseTestCase):
+
+    file_rel_path = 'submissions/closed_homework/aphacker/1/part1_buggy2.py'
+
+    def __init__(self, testname):
+        super().__init__(testname, log_in=True, user_id='ta', user_password='ta')
+
+    @unittest.skipUnless(os.environ.get('CI') is None, "cannot run in CI")
+    def get_credentials(self):
+        cookies = {k: self.driver.get_cookie(k)['value']
+                   for k in ['submitty_session', 'PHPSESSID', 'submitty_token']}
+        csrf = self.driver.find_element(By.CSS_SELECTOR, 'body').get_attribute('data-csrf-token')
+        return cookies, csrf
+
+    def test_sample_file_exists(self):
+        # this test ensures a specific sample course file we use in the subsequent test,
+        # test_delete_file, is present on the Submitty machine
+
+        self.log_out()
+        self.log_in(user_id='aphacker', user_name='aphacker', user_password='aphacker')
+        cookies, csrf = self.get_credentials()
+        endpoint = f'/courses/{self.semester}/sample/download'
+        r = requests.get(url=self.TEST_URL + endpoint, params={
+            'csrf_token': csrf,
+            'dir': 'submissions',
+            'path': f'/var/local/submitty/courses/{self.semester}/sample/' + self.file_rel_path,
+        }, cookies=cookies)
+
+        # moderately fragile checks, but since we receive status 200 on failure,
+        # this is sort of the best we can do.
+        if ('test/html' in r.headers['Content-Type']) or \
+           (b'You don\'t have access to this page.' in r.content):
+            raise ValueError(f'no access to file {self.file_rel_path}, does it exist?')
+
+        self.log_out()
+
+    @unittest.skipUnless(os.environ.get('CI') is None, "cannot run in CI")
+    def test_delete_file(self):
+        # This test ensures two things:
+        # - standard upload and deletion of files on the grading rubric page work
+        # - it isn't possible to supply alternate paths to delete other files
+        # This is kept as a single test since information from the first test
+        # is needed for the second (specifically, the anon_id)
+
+        cookies, csrf = self.get_credentials()
+
+        # 1. Go to grading details page for "Grading Homework" gradeable
+        # obtain valid anon_id of peer gradeable assignment
+        self.get(f'/courses/{self.semester}/sample/gradeable/grading_homework/grading/details')
+        # accept responsibilities as grader modal popup
+        self.driver.find_element(By.ID, 'agree-button').click()
+        # click view all sections
+        self.driver.find_element(By.CSS_SELECTOR, 'a.btn:nth-child(1)').click()
```

Comment:
A more specific selector should probably be used here if possible (when testing on main the test is failing here for me)